### PR TITLE
[glslang] Provide glslang-default-resource-limits library.

### PR DESCRIPTION
**Pull request by wolfpld**
**Describe the pull request**

- What does your PR fix?

This PR makes the glslang-default-resource-limits library publicly available for linking. This is required by third party tools using glslang, in order to get the required resource limits data structure. This struct is internally used by the glslang library and its ABI does change between releases, so it's not possible to maintain this data consistently on the third party side.

- Which triplets are supported/not supported? Have you updated the CI baseline?

No change here, in comparison to what was previously available.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

Yes, with the exception of bumping the `version` field, instead of increasing the `port-version`, as this was the convention used previously in this port.

### CODE REVIEW

File: ports/glslang/CONTROL [hunk 1]
```diff
@@ -1,4 +1,4 @@
 Source: glslang
-Version: 2019-03-05-1
+Version: 2019-03-05-2
```
Comment (NancyLi1013) [inline ports/glslang/CONTROL:2]:
```suggestion
Version: 2019-03-05
Port-Version: 2
```
Now we recommend to use `Port-Version` field here. Please see https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/control-files.md#port-version

File: ports/glslang/CMakeLists-targets.patch [hunk 2]
```diff
@@ -106,3 +106,16 @@ index f918d7a..329800f 100644
...
+diff --git a/StandAlone/CMakeLists.txt b/StandAlone/CMakeLists.txt
+index 5cea53d9..43e45534 100644
+--- a/StandAlone/CMakeLists.txt
++++ b/StandAlone/CMakeLists.txt
+@@ -46,7 +46,7 @@ if(ENABLE_GLSLANG_INSTALL)
+     install(TARGETS spirv-remap
+             RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR})
+             
+-    if(BUILD_SHARED_LIBS)
++    if(BUILD_SHARED_LIBS OR TRUE)
```
Comment (NancyLi1013) [inline ports/glslang/CMakeLists-targets.patch:118]: Frankly speaking, I don't know why you add `TRUE `here.

**Review by NancyLi1013 — CHANGES_REQUESTED** [review]

File: ports/glslang/CMakeLists-targets.patch (hunk 2 above)
Comment (wolfpld) [reply ports/glslang/CMakeLists-targets.patch:118]: That's understandable and I share your concern about this. However, adding `OR TRUE` is what this port does, for whatever reason, as you can see in the already existing portion of the `CMakeLists-targets.patch`. Therefore, the above change is consistent with previous work done here. Changing this to a proper solution is out of the scope of this PR, which is about adding a missing library.

Comment (NancyLi1013) [reply ports/glslang/CMakeLists-targets.patch:118]: Thanks for your clarification.

**Review by NancyLi1013 — APPROVED** [review]

**Comment by NancyLi1013** [comment]
LGTM now, thanks for your PR @wolfpld.

**Comment by hiradyazdan** [comment]
I am not sure how relevant this is, but since this PR relates to `glslang-default-resource-limits` I was wondering it should have resovled likning the target correctly, but I'm getting this error:

```shell
error LNK2001: unresolved external symbol "struct TBuiltInResource const glslang::DefaultTBuiltInResource" (?DefaultTBuiltInResource@glslang@@3UTBuiltInResource@@B)
SDFRay4D.exe : fatal error LNK1120: 1 unresolved externals
```

If I pull the actual repository I can add this library and link the target correctly, but I am not sure if this package cmake config links the target as in `glslang::default-resource-limits` which I assume is causing this error.

**Comment by hiradyazdan** [comment]
I am not sure how relevant this is, but since this PR relates to `glslang-default-resource-limits` I was wondering it should have resovled likning the target correctly, but I'm getting this error:

```shell
error LNK2001: unresolved external symbol "struct TBuiltInResource const glslang::DefaultTBuiltInResource" (?DefaultTBuiltInResource@glslang@@3UTBuiltInResource@@B)
....exe : fatal error LNK1120: 1 unresolved externals
```

If I pull the actual repository I can add this library and link the target correctly, but I am not sure if this package cmake config links the target as in `glslang::default-resource-limits` which I assume is causing this error.

**Pull request merged**